src/contracts/BCoWFactory.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.25;
+
+import {BCoWPool} from './BCoWPool.sol';
+
+import {BFactory} from './BFactory.sol';
+import {IBFactory} from 'interfaces/IBFactory.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+
+/**
+ * @title BCoWFactory
+ * @notice Creates new BPools, logging their addresses and acting as a registry of pools.
+ */
+contract BCoWFactory is BFactory {
+  address public solutionSettler;
+
+  constructor(address _solutionSettler) BFactory() {
+    solutionSettler = _solutionSettler;
+  }
+
+  /**
+   * @inheritdoc IBFactory
+   * @dev deploys a BCoWPool instead of a regular BPool, maintains the interface
+   * to minimize required changes to existing tooling
+   */
+  function newBPool() external override returns (IBPool _pool) {
```

Comment:
i think is ok to return `IBPool`, if someone needs `IBCoWPool` methods he can always cast it afterwards imo

Reply:
not important, but, in order to avoid copying this method entirely, what's your opinion on having an internal virtual `_newBPool` (that only deploys)

Reply:
Agree, will implement it in a separate pr and update `BCoWFactory_Unit_NewBPool.test_Set_Controller` to instead ensure the `setController` method is called with an `expectCall`